### PR TITLE
rec: Better const-ness when dealing with timestamps

### PR DESCRIPTION
**Pull request by rgacogne**
### Short description

This PR should have no functional impact, it just makes reviewing the code easier. It also saves a `gettimeofday()` call in `DTime::udiff()` which should not have any meaningful impact either.

### Checklist

I have:
- [x] read the [CONTRIBUTING.md](https://github.com/PowerDNS/pdns/blob/master/CONTRIBUTING.md) document
- [x] compiled this code
- [ ] tested this code
- [ ] included documentation (including possible behaviour changes)
- [ ] documented the code
- [ ] added or modified regression test(s)
- [ ] added or modified unit test(s)

### CODE REVIEW

File: pdns/misc.hh [hunk 1]
```diff
@@ -215,19 +219,17 @@ inline void DTime::set()
   gettimeofday(&d_set,0);
```
Comment (omoerbeek) [inline pdns/misc.hh:219]:
```suggestion
  gettimeofday(&d_set, nullptr);
```

File: pdns/misc.hh [hunk 2]
```diff
@@ -196,8 +196,12 @@ public:
   DTime & operator=(const DTime &dt) = default;
   time_t time();
```
Comment (omoerbeek) [inline pdns/misc.hh:197]: can be const and inline instead of in `time.cc` file

**Review by omoerbeek — APPROVED** [review]
Further small nits:

`getTimeval()` can be const
d_set can be initialized using {0, 0} and the default ct can be `= default`

**Comment by rgacogne** [comment]
I pushed a new commit with the changes you suggested, thanks a lot!

**Pull request merged**